app/details/[ticket_id]/ticketDetails.tsx

```diff
@@ -154,17 +156,34 @@ function TicketDetails(props: Props) {
     loadData();
   }, [loadData]);
 
-  const scrollToBottom = () => {
-    if (messagesEndRef.current) {
-      messagesEndRef.current.scrollIntoView({
-        block: 'end',
-      });
-    }
+  const handleScroll = (e: Event) => {
+    const element = e.target as HTMLDivElement;
+    const atBottom =
+      element.scrollHeight - element.scrollTop <= element.clientHeight + 50;
+    setIsUserScrolling(!atBottom);
   };
 
   useEffect(() => {
-    scrollToBottom();
-  }, [messages]);
+    const element = messagesEndRef.current?.parentElement;
+    if (!element) return;
+
+    const scrollToBottom = () => {
+      if (isUserScrolling) return;
+      element.scrollTop = element.scrollHeight;
+    };
+    const resizeObserver = new ResizeObserver(() => {
+      scrollToBottom();
+    });
+    resizeObserver.observe(element);
+
+    setTimeout(() => scrollToBottom(), 0);
+
+    element.addEventListener('scroll', handleScroll as EventListener);
```

Comment:
Remove `as EventListner`, it's redundant.

Reply:
This is use for scroll so we can't remove it @vatsal-pixer